server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -86,6 +91,9 @@ public ReindexRequest(StreamInput in) throws IOException {
         super(in);
         destination = new IndexRequest(in);
         remoteInfo = in.readOptionalWriteable(RemoteInfo::new);
+        if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
```

Comment:
I think this is going to have to be 8.0.0 until this gets backported

Reply:
Yep, you're right, changed it and will follow up after backport